xonsh/aliases.py

```diff
@@ -361,16 +376,20 @@ def which(args, stdin=None):
         print('{} -> {}'.format(match, builtins.aliases[match]))
         if pargs.all:
             try:
-                subprocess.run(['which'] + args,
-                               stderr=subprocess.PIPE,
-                               check=True)
+                subprocess.check_call(['which'] + args,
+                                      stdout=subprocess.PIPE,
+                                      stderr=subprocess.PIPE)
+                subprocess.call(['which'] + args,
+                               stderr=subprocess.PIPE)
```

Comment:
I don't grok why you have these two calls.  Would you mind adding a comment explaining it? Thanks!